src/bot/commands/ignore.ts

```diff
@@ -0,0 +1,193 @@
+import {
+  ButtonStyle,
+  ChatInputCommandInteraction,
+  ComponentType,
+  InteractionReplyOptions,
+  InteractionResponse,
+  Message
+} from 'discord.js';
+import Command from '../lib/command';
+import { Categories, CategoriesRefs, Category } from '../../typings';
+import { commandsChannelId } from '../constants';
+import { info, interactionError, interactionInfo, interactionValidate, interactionWaiter, isGodfather } from '../utils';
+import prisma from '../../prisma';
+import type { Godfather } from 'prisma/prisma-client';
+
+type MinimalGodfatherPayload = Pick<Godfather, 'id' | 'ignored_categories'> & {
+  ignored_categories: Category[];
+};
+
+export default class IgnoreCommand extends Command {
+  private static ActionButtonsMapping: Record<string, string> = {
+    discard: 'Annuler',
+    save: 'Enregistrer'
+  };
+
+  public constructor() {
+    super({
+      name: 'ignore',
+      description: 'Ouvre un menu permettant de choisir quelles catégories de blagues sont ignorées.',
+      channels: [commandsChannelId]
+    });
+  }
+
+  public async run(interaction: ChatInputCommandInteraction<'cached'>): Promise<void | InteractionResponse> {
+    let godfather: MinimalGodfatherPayload;
+    try {
+      godfather = await this.checkIsGodfather(interaction);
+    } catch (e: unknown) {
+      if (e instanceof Error) {
+        return interaction.reply(interactionInfo(e.message));
+      }
+
+      return;
+    }
+
+    const message = (await interaction.reply({
+      ...this.generateMessageOptions(godfather.ignored_categories),
+      ephemeral: true,
+      fetchReply: true
+    })) as unknown as Message<true>;
+
+    let toggledCategories: Category[] = godfather.ignored_categories;
+
+    // Ptet refactor la boucle et le interactionWaiter dans un util si jamais on a besoin de ré-implementer un menu dans le style
+    while (true) {
+      let buttonInteraction;
+      try {
+        buttonInteraction = await interactionWaiter(
+          {
+            component_type: ComponentType.Button,
+            message,
+            user: interaction.user
+          },
+          { reject_on_idle: true }
+        );
+      } catch (e: unknown) {
```

Comment:
```suggestion
      } catch (e) {
```

inutile de typer la variable e vu qu'elle n'est pas utilisée 🤔